app/src/editor/canvas/components/CodeEditor/index.jsx

```diff
@@ -1,101 +1,66 @@
-import React from 'react'
+import React, { useState, useCallback } from 'react'
 
 import CodeEditorWindow from './CodeEditorWindow'
 import DebugWindow from './DebugWindow'
+import useLayoutState from '$editor/shared/hooks/useLayoutState'
 
-export default class CodeEditor extends React.Component {
-    state = {
-        editorOpen: false,
-        debugOpen: false,
-        editorPosition: {
-            x: 0,
-            y: 0,
-        },
-        debugPosition: {
-            x: 0,
-            y: 0,
-        },
-    }
+export const CodeEditor = ({
+    children,
+    code,
+    readOnly,
+    onApply,
+    onChange,
+    debugMessages,
+    onClearDebug,
+}) => {
+    const [editorOpen, setEditorOpen] = useState(false)
+    const [debugOpen, setDebugOpen] = useState(false)
+    const [editorLayout, setEditorSize, setEditorPosition] = useLayoutState()
+    const [debugLayout, setDebugSize, setDebugPosition] = useLayoutState()
 
-    onShowEditor = () => {
-        this.setState({
-            editorOpen: true,
-        })
-    }
+    const onShowEditor = useCallback(() => {
+        setEditorOpen(true)
+    }, [])
```

Comment:
I know that @mondoreale just suggested removing the `setState` calls, and I've flip-flopped on this a few times but I would (currently) argue that they should probably be in there anyway, simply for consistency and "don't make me think". If the rule is just "list everything", then it's simple, minimal thinking required, and means you don't have to change deps if you later wrap the `setState` in a custom hook. Low impact either way so whatever, but yeah just my thoughts on this particular bike shed.

Reply:
Ok, i would agree with that - we can keep it as a "best practise" and add them always.